scripts/build.py

```diff
@@ -1393,4 +1394,4 @@ def build():
 # The 'no coverage' pragma is used as this line is un-testable. This is because
 # it will only be called when build.py is used as a script.
 if __name__ == '__main__':  # pragma: no cover
-    build()
+    main(argv=sys.argv[1:])
```

Comment:
Why skip the first argument (program name)? `parse_args` should work correctly even with it there.

Reply:
No its not working then. Its saying unrecognised args: 'file_name.py'

Reply:
It works fine locally (passing None as the value of argv), sounds like you're making an extra call? Could you provide a stack trace?

Reply:
yes it works fine with none but not with only sys.argv. Its reading an extra argument(the filename). I guess this is expected?

Reply:
That's right. Do not pass `sys.argv`, pass `None` instead.

Reply:
Wait now I am confused. If we call this script (build.py) directly from terminal with some args then the main function will get executed. But if the we are passing None to main always, then those args are never gonna get picked up, right? Instead, I think we should pass sys.argv to main here and in the definition of main, we should pass None. Something like:
```python
def main(argv=None):
    # Code.

if __name__ == '__main__':
    main(argv=sys.argv[1:])
```

Reply:
No, we are passing `None` because that is the default value `parse_args` expects. The [docs](https://docs.python.org/2/library/argparse.html#argparse.ArgumentParser.parse_args) for it explain:
> [args](https://docs.python.org/2/library/argparse.html#args) - List of strings to parse. The default is taken from [`sys.argv`](https://docs.python.org/2/library/sys.html#sys.argv).

So the flow is this.

```python
if __name__ == '__main__':
    main() # argv will be None

def main(argv=None):
    parser.parse_args(args=argv) # args is None as well, sys.argv will be checked.
```

Passing `sys.argv` directly does the wrong thing as you've already noticed, so we pass `None` instead so that `parse_args` can do the right thing.

Reply:
As a side-note, let's replace `argv` with `args` to make this less confusing.

Reply:
Done